### PR TITLE
tui: 'guicursor': use DECSCUSR for xterm-likes

### LINKED ISSUE #7550: guicursor not working
- `nvim --version`:
```
NVIM v0.2.1
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/build/config -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/src -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/opt/gettext/include -I/usr/include -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/build/src/nvim/auto -I/tmp/neovim-20171109-5288-goamgr/neovim-0.2.1/build/include
Compiled by brew@HighSierra-2.local
Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"
   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.2.1/share/nvim"
Run :checkhealth for more info
```

- Operating system/version: macOS/HighSierra
- Terminal name/version: I tried alacritty/0.1.0 and macOS builtin terminal
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC -c 'set guicursor=n:ver25'
```

### Actual behaviour
cursor shape was not changed

### Expected behaviour
change cursor shape to vertical bar

### PR DESCRIPTION
**Pull request by justinmk**
Anything claiming to be an xterm gets DECSCUSR. This is the only
reasonable choice unless/until we get more reliable detection (#7490).

ref #6997
closes #7550

cc @DarkDefender

### CODE REVIEW

**Comment by DarkDefender** [comment]
I have nothing to add except that this will break older VTE terminals that doesn't support DECSCUSR but reports that they are xterm. This applies to other outdated terminals of course, but to me it seems like most people are stuck on old VTE terminals than say older urxvt versions for example.

Maybe add a check to make sure that this doesn't trigger if `(vte_version < 3900)`? (Note that `vte_version` is 0 if no env variable for vte detection exists)

**Comment by justinmk** [comment]
@DarkDefender good point updated.

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1470,9 +1468,8 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
         // per analysis of VT100Terminal.m
         || iterm || iterm_pretending_xterm
         || teraterm    // per TeraTerm "Supported Control Functions" doco
-        // Allows forcing the use of DECSCUSR on linux type terminals, such as
-        // console-terminal-emulator from the nosh toolset, which does indeed
-        // implement the xterm extension:
+        // Some linux-type terminals (such as console-terminal-emulator
+        // from the nosh toolset) implement implement the xterm extension.
```
Comment (DarkDefender) [inline src/nvim/tui/tui.c]: Perhaps a bit late now, But there is a typo here: "implement implement"

Comment (justinmk) [reply src/nvim/tui/tui.c]: thanks, fixed in "doc" PR.

**Pull request merged**